### PR TITLE
reference hsts as best practice

### LINKED ISSUE #3189: 301 redirect for http to https susceptible to a man in the middle attacks
Re: [Switching Sites from HTTP to HTTPS](https://pantheon.io/docs/http-to-https/#redirect-to-https-and-the-primary-domain)

Specifically, Redirect to HTTPS and the primary domain

## Issue Description
We recommend a `301` redirect from HTTP to HTTPS, which is susceptible to a man in the middle attacks on HTTP requests.

## Suggested Resolution
Highly suggest setting [HSTS](https://pantheon.io/docs/hsts/) by adding it as a step to this doc. A possible alternative is to update the redirect recommendation to `307` from `301`.

### PR DESCRIPTION
**Pull request by alexfornuto**
Closes #3189 

## Effect
PR includes the following changes:
- Lists HSTS above redirect instructions as the best practice.

## Remaining Work
- [ ] Review from @ataylorme 
- [ ] Copy review

### CODE REVIEW

**Comment by ataylorme** [comment]
Looks good to me. @Shelob9 good by you?

File: source/_docs/http-to-https.md [hunk 1]
```diff
@@ -114,7 +114,11 @@ Use the following techniques to replace insecure references to your domain in th
 Clear Drupal and WordPress object caches in the [database and/or in Redis](/docs/drupal-redis/#clear-cache) in addition to manually flush edge caches by going to your Pantheon Dashboard and clicking the **Clear Caches** button.
 
 At this point, all visitors to the site should be able to securely access all pages over HTTPS with no browser warnings.
+
 ## Redirect to HTTPS and the primary domain
+
+As part of best security practices, we suggest you [Require HTTPS with the HSTS Header](/docs/hsts), instead of using a redirect.
+
 Configure redirects to the primary domain with HTTPS in `settings.php` or `wp-config.php` if a visitor arrives either (1) on a different domain or (2) without HTTPS.
```
Comment (Shelob9) [inline source/_docs/http-to-https.md]: This should be elaborated on. IE what changes in wp-config.php should be made.

Comment (alexfornuto) [reply source/_docs/http-to-https.md]: @Shelob9 do you feel that the linked guide doesn't properly cover how to enable HSTS?

Comment (Shelob9) [reply source/_docs/http-to-https.md]: I do not.

Pantheon's standard wp-config.php includes a 301 redirect for insecure requests. I am confused if you are recommending that after installing that plugin you recommend I remove the 301 redirect, change it to a 307 or what. Specific changes to wp-config.php should be detailed here.

This is contained in the phrase `Configure redirects` I suggest explaining how to configure a redirect.

**Comment by alexfornuto** [comment]
I spoke with @davidstrauss who 👍 'd this adjustment.

**Pull request merged**